portal-impl/src/com/liferay/portal/spring/context/PortalContextLoaderListener.java

```diff
@@ -329,10 +328,10 @@ public void contextInitialized(ServletContextEvent servletContextEvent) {
 		try {
 			if (PropsValues.UPGRADE_DATABASE_AUTO_RUN) {
 				DBUpgrader.upgrade(applicationContext);
-
-				StartupHelperUtil.setUpgrading(false);
 			}
 			else {
+				DBUpgrader.waitForLocks();
```

Comment:
@LuisOrtiz89 what is this for? If it is using upgrade client, it won't get to here to do upgrade. So we must be only talking about PropsValues.UPGRADE_DATABASE_AUTO_RUN case in this change, right?
I get the point of people may concurrently starting cluster nodes without external coordination and we need to support it. But all cluster nodes are having identical settings is a must do, otherwise we would fail massively one way or another.

There should not be a case where first starting node goes through  PropsValues.UPGRADE_DATABASE_AUTO_RUN upgrade, while a second node starting without it. That would be a fatal misconfiguration, we should not try to support this case.